src/bp/ui-admin/src/telemetry/middleware.ts

```diff
@@ -0,0 +1,45 @@
+import axios from 'axios'
+import _ from 'lodash'
+import ms from 'ms'
+
+import api from './../api'
+import { axiosConfig } from './client'
+
+const serverUrlPayloads = '/admin/telemetry-payloads'
+const serverUrlFeedback = '/admin/telemetry-feedback'
+
+const sendServerPackage = async () => {
+  try {
+    while (true) {
```

Comment:
while true ? where is your breaking condition?

If I understand this correctly, you fetch the data and send the packaged indefinitely until there's nothing else to send, is that correct ? 

What if theres an error posting packages to telemetry endpoint or back to botpress server ? your while loop will just run indefinitely and suck up all the browser's resources.

Please rethink this logic and make this piece of code a little cleaner by splitting into functions ... my eyes are bleeding.

Reply:
Fixed in : e135fcf